src/Files.App/UserControls/Pane/PreviewPane.xaml

```diff
@@ -294,45 +294,45 @@
 						</GridView.ItemsPanel>
 
 						<GridView.ItemTemplate>
-							<DataTemplate x:Name="TagTemplate" x:DataType="tags:TagViewModel">
-								<StackPanel
-									AutomationProperties.Name="{x:Bind Name, Mode=OneWay}"
-									Orientation="Horizontal"
-									Spacing="8"
-									ToolTipService.ToolTip="{x:Bind Name, Mode=OneWay}">
-									<PathIcon Data="{StaticResource ColorIconFilledTag}" Foreground="{x:Bind Color, Mode=OneWay}" />
-									<TextBlock
-										HorizontalAlignment="Stretch"
-										Style="{StaticResource Local.FileDetailsTextBlockStyle}"
-										Text="{x:Bind Name, Mode=OneWay}"
-										TextTrimming="CharacterEllipsis" />
+							<DataTemplate x:Name="TagTemplate" x:DataType="items:TagsListItem">
+								<StackPanel>
+									<StackPanel
+										x:Name="DetailsTagItem"
+										x:Load="{x:Bind IsTag, Mode=OneWay}"
+										AutomationProperties.Name="{x:Bind AsTag.Tag.Name, Mode=OneWay}"
+										Orientation="Horizontal"
+										Spacing="8"
+										ToolTipService.ToolTip="{x:Bind AsTag.Tag.Name, Mode=OneWay}">
+										<PathIcon Data="{StaticResource ColorIconFilledTag}" Foreground="{x:Bind AsTag.Tag.Color, Mode=OneWay}" />
+										<TextBlock
+											HorizontalAlignment="Stretch"
+											Style="{StaticResource Local.FileDetailsTextBlockStyle}"
+											Text="{x:Bind AsTag.Tag.Name, Mode=OneWay}"
+											TextTrimming="CharacterEllipsis" />
+									</StackPanel>
+
+									<!--  Edit Tags Button  -->
+									<Button
+										x:Name="DetailsEditTags"
+										HorizontalAlignment="Left"
+										x:Load="{x:Bind IsFlyout, Mode=OneWay}"
+										AutomationProperties.Name="{helpers:ResourceString Name=EditTags}"
+										Flyout="{x:Bind AsFlyout.Flyout, Mode=OneWay}"
+										ToolTipService.ToolTip="{helpers:ResourceString Name=EditTags}">
+										<Button.Content>
+											<FontIcon FontSize="12" Glyph="&#xE710;" />
```

Comment:
Perhaps we can still use the tag icon
`<PathIcon Data="{StaticResource ColorIconFilledTag}" Foreground="{x:Bind Color, Mode=OneWay}" />`

Reply:
Do you mean instead of the `+`?

Reply:
Yes, and I like that icon better, good idea.

Reply:
@ferrariofilippo I added the icon to PathIcons.xaml `ColorIconTags`.

Reply:
![image](https://github.com/files-community/Files/assets/102259289/d2935e06-751a-445d-b6cc-e1194b6fb1dc)

I think it needs an outline

Reply:
Fixed